### DISCUSSION TITLE
Near misses on many common backyard birds - why?

### DISCUSSION THREAD

**Discussion by jflyons** [mcguirepr89/BirdNET-Pi#580] [General]
Forgive my newbie perspective, as I have had my BirdNET-Pi going just about six weeks, with some exciting results, but also some puzzling outcomes. It's proving great at recording our neighborhood Western Screech-Owls, but misses some obvious sounds (I can hear live, and see on the Spectrogram) - these include HOFI, LEGO, AMCR - when I watch they log they sometimes get in the .3-.4 range, occasionally higher, and ever so rarely over .7. I have used Merlin Sounds app for over a year on a daily basis, sometimes BirdNET too, and they are normally very quick to pick up the aforementioned, as well as (as I just heard) the chip notes of the Yellow-rumped Warbler (which are currently moving through in large numbers).

I have tried numerous microphone solutions - again, many with quite good results on some species. But missing those I know I would be picking up via my other devices (tablet and iPhone) and the mobile apps.

Thanks in advance for any help and I am prepared to be humbled by a simple answer I have overlooked, or a technical one which I am willing to take on!

**Comment by ehpersonal38** [comment]
Hi @jflyons, glad to hear you're having mostly successes with your BirdNET-Pi!

I do suspect that the low confidence behavior you're seeing is due to the fact that BirdNET-Pi is currently using a model that was last updated about 2 years ago whereas the BirdNET mobile app (and I'm sure Merlin too) gets regular model updates. Patrick has begun work as of last month on [implementing the new model](https://github.com/mcguirepr89/BirdNET-Analyzer-Pi), though I suspect it won't be fully ready for another several months. So for now I'm not sure there's much that can be done for your installation, other than maybe adding in some code that will lower the minimum confidence interval on a species-by-species basis. Something like (psuedocode):

```
if species == "American Crow" and confidence > 0.3:
--- log detection
```

Of course it's not a perfect solution, but let me know if you're interested in implementing that manually and I can offer some additional guidance. 

🐤

**Reply by jflyons** [reply]
Thanks greatly - the insight into model updates makes sense. 

The coding is more than I have done here, but would welcome your further guidance - thanks again!

**Reply by ehpersonal38** [reply]
You're welcome!

So the file we're going to be modifying for this is `~/BirdNET-Pi/scripts/server.py`. You can do a find and replace for 
```python
entry[1] >= min_conf
```
and replace it with 

```python
(entry[1] >= min_conf or (entry[0].split("_")[1] in species_confidence_overrides and entry[1] >= species_confidence_overrides[entry[0].split("_")[1]]) )
```

Then you can add this variable around line 27 of the same file:
```python
species_confidence_overrides = {"American Crow": 0.2, "House Finch": 0.3}
```

Feel free to tweak that as needed to set the minimum confidence for each species as desired. When you're ready to apply the changes, save the file and restart all services with `bash ~/BirdNET-Pi/scripts/restart_services.sh`

Hoping this works for you! Do let me know how it goes.

**Reply by RiccoRammer** [reply]
I gave this a try and it worked perfectly, I would prefer to have the override with the scientific name
Would this work?
`(entry[1] >= min_conf or (entry[0].split("_")[0] in species_confidence_overrides and entry[1] >= species_confidence_overrides[entry[0].split("_")[1]]) )`

`species_confidence_overrides = {"Numida meleagris": 0.04}`

**Reply by RiccoRammer** [reply]
This worked for the scientific name
`(entry[1] >= min_conf or (entry[0].split("_")[0] in species_confidence_overrides and entry[1] >= species_confidence_overrides[entry[0].split("_")[0]]) )`

and
`species_confidence_overrides = {"Numida meleagris": 0.04}`
